src/python/pants/backend/go/util_rules/build_pkg.py

```diff
@@ -752,10 +752,7 @@ async def build_go_package(
     # about the Go code that can be used by assembly files.
     asm_header_path: str | None = None
     if s_files:
-        if os.path.isabs(request.dir_path):
-            asm_header_path = "go_asm.h"
-        else:
-            asm_header_path = os.path.join(request.dir_path, "go_asm.h")
+        asm_header_path = str(PurePath("go_asm.h"))
```

Comment:
Maybe put the `go_asm.h` file into the `obj_dir_path` (if one exists)? (The existing code kept it scoped into the package's source directory, i.e. `request.dir_path`, but it is a generated file and should probably be treated like the object files built from assembly sources and kept in `obj_dir_path`.) Thoughts?

Reply:
I originally tried using `_objdir/go_asm.h`, but the builds started failing for standard libraries. I spent an unsuccessful time looking for other references to the go_asm.h path inside of pants, so I'm not sure why those library builds did not find it in _objdir. 

Long story - I'd rather leave it as is, because the fix has some complexity to it, but I'd be happy to add a comment or file a follow up task.

Reply:
Please open an issue then with some details on the fix.

Reply:
Filed [#20618](https://github.com/pantsbuild/pants/issues/20618) with what I know.